odoo/fields.py

```diff
@@ -660,6 +660,37 @@ def get(key, get_context=env.context.get):
 
         return tuple(get(key) for key in self.depends_context)
 
+    def mapped(self, records):
+        if self.name == 'id':
+            # not stored in cache
+            return records._ids
+
+        if self.compute:
+            # a recompute must be forced for records to_recompute that intersect with input records
+            # otherwise an infinite loop will occur on the next call to flush() / recompute()
+            # as the value will be False in cache and will never trigger a MissingError which,
+            # in turn, triggers a recompute
+            to_compute_ids = records.env.all.tocompute.get(self, set())
+            if not to_compute_ids.isdisjoint(records._ids):
+                to_compute_ids = to_compute_ids.intersection(records._ids).difference(
+                    records.env._protected.get(self, ()))
+                self.compute_value(records.browse(to_compute_ids.intersection(records._ids)))
```

Comment:
Why intersecting again with `records._ids`?
Note also that `to_compute_ids` may be empty once protected ids are subtracted.

Why not doing:
```python
to_compute_ids = records.env.all.tocompute.get(self, ())
to_protect_ids = records.env._protected.get(self, ())
ids = [id_ for id_ in record._ids
       if id_ in to_compute_ids and id_ not in to_protect_ids]
if ids:
    self.compute_value(records.browse(ids))
```

Reply:
Wait, we can simplify this!  By design, a field cannot be both to compute and protected on a given record.  So you can actually ignore protected fields here...

Reply:
/!\ Recursive fields must be computed one by one; while the records `to_compute` and `records` intersect, we must compute the first record of the recordset